### ISSUE TITLE
IsEquivalentTo wrongly passes when comparing a dictionary with a similar subset missing some keys

### ISSUE THREAD

**Issue by tpehkone** [tpierrain/NFluent#306]
## Expected Behavior
When comparing dictionaries, IsEquivalentTo() should fail if one dictionary is missing keys present in the other. It should not matter which way around the dictionaries are compared.

## Current Behavior
IsEquivalentTo() falsely passes, depending on which way the comparison is done.

## Code that reproduce the issue 

```csharp
[Test]
public void Test_Dict_IsEquivalentTo()
{
    var dictA = new Dictionary<string, string> { { "aa", "AA" }, { "bb", "BB" } };
    var dictB = new Dictionary<string, string> { { "aa", "AA" } };
    Check.ThatCode(() => Check.That(dictB).IsEquivalentTo(dictA)).Throws<AssertionException>();  // Fails, as should.
    Check.ThatCode(() => Check.That(dictA).IsEquivalentTo(dictB)).Throws<AssertionException>();  // Should fail, does not!
}
```

## Your Environment

* Version used: 2.6.0
* Framework version: .Net Core 2.1

**Comment by tpehkone** [comment]
This seems to work fine now in my own tests, too.

**Comment by dupdob** [comment]
fixed in V 2.7.0
